File: sdklibrary/src/main/java/com/example/cue/sdklibrary/apiModel/RetrofitClientInstance.java
package com.example.cue.sdklibrary.apiModel;

import java.util.concurrent.TimeUnit;

import okhttp3.CertificatePinner;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClientInstance {

    private static Retrofit retrofit;
    private static final String BASE_URL = "https://connect-staging-20.blackline-dev.com";
    // Staging key
    private static final String sha_256 = "yup3ktLnGP2tW4U40xlLhT6RzUPHzAlUlM8pekl+udE=";
    private static final String hostname = "connect-staging-20.blackline-dev.com";
    // Production key
  //  private static final String sha_256 = "OkCcMYBVBnbpc8KRJbdVytXdu1BNZy6y4jKgK97PfzY=";
  //  private static final String hostname = "connect-live.blacklinesafety.com";
    private static final int TIMEOUT = 60000;


    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

            CertificatePinner certificatePinner = new CertificatePinner.Builder()
                    .add(hostname, "sha256/"+sha_256)
                    .build();

            OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor)
                    .certificatePinner(certificatePinner)
                    .connectTimeout(TIMEOUT, TimeUnit.MILLISECONDS).build();

            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(client)
                    .build();
        }
        return retrofit;
    }
}

File: LonerSdkLibrary/src/main/java/com/loner/android/sdk/webservice/network/networking/RetrofitClientInstance.kt
package com.loner.android.sdk.webservice.network.networking

import okhttp3.CertificatePinner
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

/**
 * <p>This is singleton class, which provide a Retrofit object.
 *    It has also implemented certificatePinning for api calling.
 */
class RetrofitClientInstance : RequestConfig {
    companion object {
        private var retrofit: Retrofit? = null
        val retrofitInstance: Retrofit?
            get() {
                if (retrofit == null) {
                    val interceptor = HttpLoggingInterceptor()
                    interceptor.level = HttpLoggingInterceptor.Level.BODY

                    val certificatePinner = CertificatePinner.Builder()
                            .add(RequestConfig.HOST_NAME, "sha256/" + RequestConfig.SHA_256)
                            .build()

                    val client = OkHttpClient.Builder().addInterceptor(interceptor)
                            .certificatePinner(certificatePinner)
                            .callTimeout(RequestConfig.TIMEOUT.toLong(),TimeUnit.SECONDS).build()


                    retrofit = Retrofit.Builder()
                            .baseUrl(RequestConfig.BASE_URL)
                            .addConverterFactory(GsonConverterFactory.create())
                            .client(client)
                            .build()
                }
                return retrofit
            }
    }
}
